include/dlaf/matrix.h

```diff
@@ -14,6 +14,7 @@
 #include "dlaf/communication/communicator_grid.h"
 #include "dlaf/matrix/distribution.h"
```

Comment:
I don't know what you think about it. IMHO Matrix does not depend anymore on Distribution (directly), so I would remove it.